script.js

```diff
@@ -1,4 +1,246 @@
+      cards.push(card);
+
+      // Increment rank Counter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cards;
 };
+
+var makeDeckDone = makeDeck();
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cards array
+function shuffleDeck(cards) {
+  // Loop over the card deck array once
+  var index = 0;
+  while (index < cards.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cards.length);
+
+    // Select the card that corresponds to randomIndex
+    var randomItem = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentItem = cards[index];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[index] = randomItem;
+
+    cards[randomIndex] = currentItem;
+    // Increment currentIndex
+    index += 1;
+  }
+  // Return the shuffled deck
+  return cards;
+}
+
+// Function to Create NEW DECK
+
+function createNewDeck() {
+  var shuffleDeckDone = shuffleDeck(makeDeckDone);
+  return shuffleDeckDone;
+}
+
+//////// got blackjack?
+
+function gotBlackjack(handArray) {
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+
+  var gotBlackjack = false;
+
+  if (
+    (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||
+    (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
+  ) {
+    gotBlackjack = true;
+  } else {
+    gotBlackjack = false;
+  }
+  return gotBlackjack;
+}
+
+function sumHandScore(handArray) {
+  var sum = 0;
+  var i = 0;
+
+  while (i < handArray.length) {
+    var currentCard = handArray[i];
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      sum = sum + 10;
+    } else {
+      sum = sum + currentCard.rank;
+    }
+    i += 1;
+  }
+  return sum;
+  ``;
+}
+
+// Game MODES
+var gameStart = "game start";
+var gameCardsIssued = "cards issued";
+var gameScore = "game score";
+var gameModeNow = gameStart;
+var playDeck = "";
+
+//////////////////////////////////////////
+
+function main(input) {
+  var instruction;
+
+  // SUBMIT
+  if (gameModeNow == gameStart) {
+    // create the game deck
+
+    playDeck = createNewDeck();
+    console.log(playDeck);
+
+    // deal 2 cards to player and dealer respectively
+
+    playerHand.push(playDeck.pop());
+    dealerHand.push(playDeck.pop());
+    playerHand.push(playDeck.pop());
+    dealerHand.push(playDeck.pop());
+
+    console.log("player hand >>");
+    console.log(playerHand);
+    console.log("dealer hand >>");
+    console.log(dealerHand);
+
+    gameModeNow = gameCardsIssued;
+
+    instruction = "Cards are dealt. Press SUBMIT to see cards";
+
+    return instruction;
```

Comment:
Do you think you could try to display the cards to the user instead of just console.log ?